pallets/foreign-investments/src/impls/invest.rs

```diff
@@ -0,0 +1,967 @@
+					swap: Swap {
+						amount: swap.amount.ensure_add(pool_swap.amount)?,
+						..swap
+					},
+				})
+			}
+			// Reduce foreign swap amount by the increasing amount and increase investing amount as
+			// well adding foreign_done amount by the minimum of active swap amounts
+			Self::ActiveSwapIntoForeignCurrency { swap: foreign_swap } => {
+				swap.ensure_currencies_match(foreign_swap, false)?;
+				let invest_amount = swap.amount.min(foreign_swap.amount);
+				let done_amount = swap.amount.min(foreign_swap.amount);
+
+				match swap.amount.cmp(&foreign_swap.amount) {
+					// pool swap amount is immediately invested and done amount increased equally
+					Ordering::Equal => {
+						Ok(
+							Self::ActiveSwapIntoForeignCurrencyAndSwapIntoForeignDoneAndInvestmentOngoing {
+								swap: Swap {
+									// safe since swap.amount < foreign_swap.amount
+									amount: foreign_swap.amount - swap.amount,
+									..*foreign_swap
+								},
+								done_amount,
+								invest_amount,
+							},
+						)
+					}
+					// swap amount is immediately invested and done amount increased equally
+					Ordering::Less => Ok(Self::SwapIntoForeignDoneAndInvestmentOngoing {
+						done_swap: *foreign_swap,
+						invest_amount,
+					}),
+					// foreign swap amount is immediately invested and done amount increased equally
+					Ordering::Greater => {
+						Ok(
+							Self::ActiveSwapIntoPoolCurrencyAndSwapIntoForeignDoneAndInvestmentOngoing {
+								swap: Swap {
+									// safe since swap.amount > foreign_swap.amount
+									amount: swap.amount - foreign_swap.amount,
+									..swap
+								},
+								done_amount,
+								invest_amount,
+							},
+						)
+					}
+				}
+			}
+			// Bump pool swap
+			Self::ActiveSwapIntoPoolCurrencyAndInvestmentOngoing {
+				swap: pool_swap,
+				invest_amount,
+			} => {
+				swap.ensure_currencies_match(pool_swap, true)?;
+
+				Ok(Self::ActiveSwapIntoPoolCurrencyAndInvestmentOngoing {
+					swap: Swap {
+						amount: swap.amount.ensure_add(pool_swap.amount)?,
+						..swap
+					},
+					invest_amount: *invest_amount,
+				})
+			}
+			// Reduce foreign swap amount by the increasing amount and increase investing amount as
+			// well adding foreign_done amount by the minimum of active swap amounts
+			Self::ActiveSwapIntoForeignCurrencyAndInvestmentOngoing {
+				swap: foreign_swap,
+				invest_amount,
+			} => {
+				swap.ensure_currencies_match(foreign_swap, false)?;
+				let invest_amount =
+					invest_amount.ensure_add(swap.amount.min(foreign_swap.amount))?;
+				let done_amount = swap.amount.min(foreign_swap.amount);
+
+				match swap.amount.cmp(&foreign_swap.amount) {
+					// pool swap amount is immediately invested and done amount increased equally
+					Ordering::Equal => {
+						Ok(
+							Self::ActiveSwapIntoForeignCurrencyAndSwapIntoForeignDoneAndInvestmentOngoing {
+								swap: Swap {
+									// safe since swap.amount < foreign_swap.amount
```

Comment:
Same here

Reply:
Thanks, while working on fixing amount denominations, I realized this section was off. Luckily, it does not affect same currency investments.